### ISSUE TITLE
Bug: Incompatibility with Calendarium Plugin

### ISSUE THREAD

**Issue by namtrah** [namikaze-40p/obsidian-date-inserter#9]
Hello, When I have your plugin and the Calendarium plugin from Javalent installed, your plugin shows weird elements that are part of the other plugins span elements. It seems your styles are not nested properly and their styling overrides yours making it ineligible. 

Could you make sure the styles for date picker are nested so they only apply to the modal according to the vanilla-datepicker docs, they should be... but the Date Inserter plugin is adding the styles, and it would appear the styles they are adding aren't scoped to the modal (so they're applying to anything with a data-date field).

See this Discord entry for more details https://discord.com/channels/686053708261228577/1286771469724160021/1317072012874743849

**Comment by namikaze-40p** [comment]
I installed Calendarium and reviewed the issue you mentioned.
While I was able to identify the cause, I believe there may not be much I can do from the Date Inserter side to address it. 

<img width="1433" alt=" img" src="https://github.com/user-attachments/assets/e0de575a-6896-4c75-98de-d98cb8e52303" />

The issue stems from a styling rule that applies to span or div tags that are descendants of a body tag without the specific class unique to Calendarium.

```css
body:not(.calendarium-nix-spans) span[data-date]:not([data-end]):not(.hide-data):before, body:not(.calendarium-nix-spans) div[data-date]:not([data-end]):not(.hide-data):before {
	content: "\1f516 " attr(data-date) ": " attr(data-name) ". ";
	color: var(--calendarium-span-data-color, var(--text-accent));
	font-weight: 500;
}
```

This rule will likely apply to any span or div tags that match the criteria, not just those from Date Inserter, but also from other plugins. I think it would be more appropriate for the Calendarium plugin to limit the scope of its unique styles to avoid such conflicts.

As a temporary solution, you can resolve the issue by applying the following [CSS snippets](https://help.obsidian.md/Extending+Obsidian/CSS+snippets). Please try it if you like.
```css
body:not(.calendarium-nix-spans) .date-inserter-modal span[data-date]:not(.hide-data),
body:not(.calendarium-nix-spans) .date-inserter-modal div[data-date]:not(.hide-data) {
	font-style: inherit;
	font-weight: inherit;
}

body:not(.calendarium-nix-spans) .date-inserter-modal span[data-date]:not([data-end]):not(.hide-data)::before,
body:not(.calendarium-nix-spans) .date-inserter-modal div[data-date]:not([data-end]):not(.hide-data)::before {
	content: "";
	color: inherit;
	font-weight: inherit;
}

body:not(.calendarium-nix-spans) .date-inserter-modal span[data-date][data-end]:not(.hide-data)::before ,
body:not(.calendarium-nix-spans) .date-inserter-modal div[data-date][data-end]:not(.hide-data)::before {
	content: "";
	color: inherit;
	font-weight: inherit;
}
```

**Comment by namikaze-40p** [comment]
I'm not entirely sure if this is the correct approach, but I was able to test a method for displaying icons using the span tag in Calendarium.

> As a temporary solution, you can resolve the issue by applying the following [CSS snippets](https://help.obsidian.md/Extending+Obsidian/CSS+snippets). Please try it if you like.

By utilizing the CSS snippets mentioned above, I confirmed that it is possible to achieve both the use of span tags in Calendarium and maintain the proper display of the calendar in Date Inserter without any issues.

https://github.com/user-attachments/assets/dcb174e9-85ac-42f3-9c5c-2f8e92d7e1ea

If no other problems arise, I'll close this issue in the near future.
